addon/components/uk-modal.js

```diff
@@ -1,58 +1,74 @@
 import { getOwner } from "@ember/application";
 import Component from "@ember/component";
-import { computed } from "@ember/object";
+import { assert } from "@ember/debug";
+import { guidFor } from "@ember/object/internals";
 import { scheduleOnce } from "@ember/runloop";
+import { tracked } from "@glimmer/tracking";
 import UIkit from "uikit";
 
 import layout from "../templates/components/uk-modal";
 
 const noop = () => {};
 
-export default Component.extend({
-  layout,
-
-  modalClass: "",
-  dialogClass: "",
-  btnClose: true,
-  escClose: true,
-  bgClose: true,
-  stack: false,
-  container: true,
-  clsPage: "uk-modal-page",
-  clsPanel: "uk-modal-dialog",
-  selClose: [
+export default class UkModal extends Component {
+  layout = layout;
+
+  modalId = `modal-${guidFor(this)}`;
```

Comment:
Might be better to use a uuid so it's future proof (glimmer)

Reply:
is glimmer not playing well with `guidFor`?

Reply:
> You can call this on any object, EmberObject-based or not.

According to the docs this only works on ember objects which [a glimmer component isn't](https://github.com/glimmerjs/glimmer.js/blob/master/packages/%40glimmer/component/addon/-private/component.ts#L142).

Reply:
EmberObject-based of **not**.

According to the api docs this is what is accepted for first argument:
> obj Object any object, string, number, Element, or primitive

https://api.emberjs.com/ember/release/functions/@ember%2Fobject%2Finternals/guidFor

Reply:
:rofl: reading helps. NVM then. Now you just have to resolve conflicts and we're good to go